### PR TITLE
Alerting: Improve test coverage for ConditionsCmd

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**Which issue(s) does this PR fix?**:

Before making further changes to `ConditionsCmd` I want to improve the test coverage to make sure we that still match the expected behaviour. There are some tests in this pull request that, while pass, do not seem to be correct, and will be investigated in a later pull request.

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by joeblubaugh — APPROVED** [review]
> Could you mark the tests that are illustrating incorrect behavior with a comment?

I did on the assertion, would it be better if it was above the name of the test?

File: pkg/expr/classic/classic_test.go [hunk 1]
```diff
@@ -57,214 +61,396 @@ func TestConditionsCmd(t *testing.T) {
...
+					InputRefID: "A",
+					Reducer:    reducer("min"),
+					Operator:   "and",
+					Evaluator:  &noValueEvaluator{},
+				},
+			},
+		},
+		expected: func() mathexp.Results {
+			v := newNumber(nil)
+			// This seems incorrect
```
Comment (grobinson-grafana) [inline pkg/expr/classic/classic_test.go:207]: @joeblubaugh

**Comment by grobinson-grafana** [comment]
@joeblubaugh going to merge this to keep making progress, will open another pull request this afternoon that we can put feedback into?

**Comment by joeblubaugh** [comment]
all good @grobinson-grafana

**Pull request merged**